libs/test-utils/src/mock_function.ts

```diff
@@ -114,22 +128,69 @@ export function mockFunction<Func extends AnyFunc>(
     actualCalls: [],
   };
 
+  // Find the next expected call that should be used for the given actual call.
+  // We track which expected call each actual call matched and then base the
+  // next appropriate expected call on that.
+  function findExpectedCallForActualCallIndex(actualCallIndex: number) {
+    const previousActualCall = state.actualCalls[actualCallIndex - 1];
+    if (!previousActualCall) {
+      return state.expectedCalls[0];
+    }
+
+    // If the previous actual call didn't match any expected call, then we don't
+    // know what the next expected call should be.
+    if (previousActualCall.matchingExpectedCallIndex === undefined) {
+      throw new MockFunctionError(
+        `Previous call to mock function did not match expected calls`
```

Comment:
nit: Can use standard quotes here instead of backticks

Reply:
Thanks, fixed